### PR TITLE
Create codeql-analysis.yml

### PR DESCRIPTION
**Pull request by sameersbn**

### CODE REVIEW

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v2
+
+    # Initializes the CodeQL tools for scanning.
+    - name: Initialize CodeQL
+      uses: github/codeql-action/init@v1
+      with:
+        languages: ${{ matrix.language }}
```
Comment (antoineco) [inline .github/workflows/codeql-analysis.yml:33]: Is that really necessary?

File: .github/workflows/codeql-analysis.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  analyze:
+    name: Analyze
+    runs-on: ubuntu-latest
+    permissions:
+      actions: read
+      contents: read
+      security-events: write
+
+    strategy:
+      fail-fast: false
```
Comment (antoineco) [inline .github/workflows/codeql-analysis.yml:21]: If the build doesn't work, no need to run the analysis.

File: .github/workflows/codeql-analysis.yml (hunk 1 above)
Comment (sameersbn) [reply .github/workflows/codeql-analysis.yml:33]: not really, it was autogenerated.

File: .github/workflows/codeql-analysis.yml [hunk 3]
```diff
@@ -0,0 +1,39 @@
+name: "CodeQL"
```
Comment (antoineco) [inline .github/workflows/codeql-analysis.yml:1]: No need for quotes around strings in YAML.

File: .github/workflows/codeql-analysis.yml [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+  schedule:
+    - cron: '23 18 * * 1'
```
Comment (antoineco) [inline .github/workflows/codeql-analysis.yml:9]: nit: No need for quotes here either (the string doesn't start with `*` or `&`).

File: .github/workflows/codeql-analysis.yml (hunk 1 above)
Comment (antoineco) [reply .github/workflows/codeql-analysis.yml:33]: Can we remove it then? This repo is unlikely to become a multi-language project.

Comment (sameersbn) [reply .github/workflows/codeql-analysis.yml]: done

**Review by antoineco — APPROVED** [review]
Approving, but `fail-fast` is definitely not what we want I think (it would only if we were running the same workflow multiple times in a matrix).

**Review by antoineco — CHANGES_REQUESTED** [review]
Hold on. We do need autobuild.
Go is a compiled language, so GraphQL will analyse nothing if we don't build (and report ok every time).

**Comment by antoineco** [comment]
Marking as draft because the PR is changing so rapidly that one review is immediately discarded.

Let's take our time and mark as ready when it's ready.

**Comment by sameersbn** [comment]
Sorry, I noticed the autobuild step was not going anything (0s spent), so I just tried removing to see what would happen. I've added it back though
I am done changing the PR, you may review it now @antoineco

**Comment by sameersbn** [comment]
well one last change, i need to put the language matrix back as it refered to by autobuild

File: .github/workflows/codeql-analysis.yml (hunk 1 above)
Comment (sameersbn) [reply .github/workflows/codeql-analysis.yml:29]: had to put it back because it was being referred by the autobuild step (check out the actions output)

**Review by antoineco — APPROVED** [review]

**Pull request merged**